frontend/packages/console-dynamic-plugin-sdk/src/app/k8s/api-discovery/api-discovery.ts

```diff
@@ -0,0 +1,174 @@
+    return `${kind}s`;
+  }
+  return pluralized;
+};
+
+const defineModels = (list: APIResourceList): K8sModel[] => {
+  const groupVersionParts = list.groupVersion.split('/');
+  const apiGroup = groupVersionParts.length > 1 ? groupVersionParts[0] : null;
+  const apiVersion = groupVersionParts.length > 1 ? groupVersionParts[1] : list.groupVersion;
+  return list.resources
+    .filter(({ name }) => !name.includes('/'))
+    .map(({ name, singularName, namespaced, kind, verbs, shortNames }) => {
+      return {
+        kind,
+        namespaced,
+        verbs,
+        shortNames,
+        label: kind,
+        plural: name,
+        apiVersion,
+        abbr: kindToAbbr(kind),
+        ...(apiGroup ? { apiGroup } : {}),
+        labelPlural: pluralizeKind(kind),
+        path: name,
+        id: singularName,
+        crd: true,
+      };
+    });
+};
+
+const getResources = async (): Promise<DiscoveryResources> => {
+  let apiResourceData;
+  try {
+    apiResourceData = await consoleFetchJSON(`${k8sBasePath}/apis`);
+  } catch (err) {
+    throw err;
+  }
+  const groupVersionMap = apiResourceData.groups.reduce(
+    (acc, { name, versions, preferredVersion: { version } }) => {
+      acc[name] = {
+        versions: _.map(versions, 'version'),
+        preferredVersion: version,
+      };
+      return acc;
+    },
+    {},
+  );
+  const all: Promise<APIResourceList>[] = _.flatten(
+    apiResourceData.groups.map((group) =>
+      group.versions.map((version) => `/apis/${version.groupVersion}`),
+    ),
+  )
+    .concat(['/api/v1'])
+    .map((p) => consoleFetchJSON(`api/kubernetes${p}`).catch((err) => err));
+
+  return Promise.all(all).then((data) => {
+    const resourceSet = new Set<string>();
+    const namespacedSet = new Set<string>();
+    data.forEach(
+      (d) =>
+        d.resources &&
+        d.resources.forEach(({ namespaced, name }) => {
+          resourceSet.add(name);
+          namespaced && namespacedSet.add(name);
+        }),
+    );
+    const allResources = [...resourceSet].sort();
+
+    const safeResources = [];
+    const adminResources = [];
+    const models = _.flatten(data.filter((d) => d.resources).map(defineModels));
+    const coreResources = new Set([
+      'roles',
+      'rolebindings',
+      'clusterroles',
+      'clusterrolebindings',
+      'thirdpartyresources',
+      'nodes',
+      'secrets',
+    ]);
+    allResources.forEach((r) =>
+      coreResources.has(r.split('/')[0]) ? adminResources.push(r) : safeResources.push(r),
+    );
+    const configResources = _.filter(
+      models,
+      (m) => m.apiGroup === 'config.openshift.io' && m.kind !== 'ClusterOperator',
+    );
+    const clusterOperatorConfigResources = _.filter(
+      models,
+      (m) => m.apiGroup === 'operator.openshift.io',
+    );
+
+    return {
+      allResources,
+      safeResources,
+      adminResources,
+      configResources,
+      clusterOperatorConfigResources,
+      namespacedSet,
+      models,
```

Comment:
we can only store models and skip others but we get all these resources with the same calls so we can have it too.

Reply:
@invincibleJai so you're saying we don't need the other resources just yet/don't store them, but you've kept them in case we do in the future?

Reply:
Oh. I think I see what you are getting at. If we don't pass all this along, the `receivedResources` action will break?

Reply:
yeah it will,  i used exsiting selector/reducers what we have.